File: backend/src/models/Insumo.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Insumo = sequelize.define('Insumo', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  emprendimientoId: {
    type: DataTypes.UUID,
    allowNull: false,
    field: 'emprendimiento_id',
  },
  proveedorId: {
    type: DataTypes.UUID,
    allowNull: true,
    field: 'proveedor_id',
  },
  nombre: {
    type: DataTypes.STRING(150),
    allowNull: false,
  },
  unidadMedida: {
    type: DataTypes.STRING(20),
    allowNull: false,
    defaultValue: 'unidad',
    field: 'unidad_medida',
  },
  stockActual: {
    type: DataTypes.DECIMAL(12, 3),
    allowNull: false,
    defaultValue: 0,
    field: 'stock_actual',
  },
  stockMinimo: {
    type: DataTypes.DECIMAL(12, 3),
    allowNull: false,
    defaultValue: 0,
    field: 'stock_minimo',
  },
  costoUnitario: {
    type: DataTypes.DECIMAL(12, 2),
    allowNull: false,
    defaultValue: 0,
    field: 'costo_unitario',
  },
  activo: {
    type: DataTypes.BOOLEAN,
    defaultValue: true,
  },
}, {
  tableName: 'insumos',
});

module.exports = Insumo;

File: mobile/lib/models/insumo.dart
import 'parsing.dart';

class Insumo {
  final String id;
  final String nombre;
  final String unidadMedida;
  final double stockActual;
  final double stockMinimo;
  final double costoUnitario;

  Insumo({
    required this.id,
    required this.nombre,
    required this.unidadMedida,
    required this.stockActual,
    required this.stockMinimo,
    required this.costoUnitario,
  });

  bool get bajoStock => stockActual <= stockMinimo;

  factory Insumo.fromJson(Map<String, dynamic> json) {
    return Insumo(
      id: json['id'] as String,
      nombre: json['nombre'] as String,
      unidadMedida: json['unidadMedida'] as String? ?? 'unidad',
      stockActual: parseDouble(json['stockActual']),
      stockMinimo: parseDouble(json['stockMinimo']),
      costoUnitario: parseDouble(json['costoUnitario']),
    );
  }
}
